File: src/main/resources/scala-native/qt/core/QCoreApplication.cpp
#include <QCoreApplication>
#include "QCoreApplication.hpp"

void q_core_application_set_organization_domain(const q_string orgDomain) {
  QCoreApplication::setOrganizationDomain(*((QString*)orgDomain));
  return;
}

void q_core_application_set_organization_name(const q_string orgName) {
  QCoreApplication::setOrganizationName(*((QString*) orgName));
  return;
}

void q_core_application_set_application_name(const q_string applicationName) {
  QCoreApplication::setApplicationName(*((QString*) applicationName));
  return;
}

File: scala-native/src/main/scala/qt/core/QCoreApplication.scala
package qt.core

import scalanative.unsafe.*
import qt.QString
import scala.util.Using
import scala.util.Using.Manager

@extern
@link("Qt5Core")
object _QCoreApplication:
  def q_core_application_set_organization_name(orgName: QString): Unit = extern
  def q_core_application_set_organization_domain(orgDomain: QString): Unit =
    extern

  def q_core_application_set_application_name(appName: QString): Unit = extern

object QCoreApplication:
  def setOrganizationDomain(orgDomain: String)(using m: Manager) =
    _QCoreApplication.q_core_application_set_organization_domain(
      QString(orgDomain)
    )

  def setOrganizationName(orgName: String)(using m: Manager) =
    _QCoreApplication.q_core_application_set_organization_name(QString(orgName))

  def setApplicationName(name: String)(using m: Manager) =
    _QCoreApplication.q_core_application_set_application_name(QString(name))
